climpred/utils.py

```diff
@@ -301,6 +340,54 @@ def _load_into_memory(res):
     return res
 
 
+def reduce_forecast_to_same_inits(forecast, verif):
+    """Reduces the forecast to common set of initializations that verify over all lags.
+
+    Args:
+        forecast (``xarray`` object): Prediction ensemble with ``init`` dim renamed to
+            ``time`` and containing ``lead`` dim.
+        verif (``xarray`` object): Verification data with ``time`` dim.
+
+    Returns:
+        forecast (``xarray`` object): Prediction ensemble with ``init`` sub-selected for
+            those that verify over all leads.
+        verif (``xarray`` object): Original verification data.
+    """
+    # Construct list of `n` offset over all leads.
+    n, freq = get_multiple_lead_cftime_shift_args(
+        forecast['lead'].attrs['units'], forecast['lead'].values
+    )
+    n = list(n)
+    # Add lead 0 to check that init exists in the observations so that reference
+    # forecasts have the same set of inits.
+    if 0 not in n:
+        n.insert(0, 0)
+    # Note that `init` is renamed to `time` in the compute function to compute metrics.
+    init_lead_matrix = xr.concat(
+        [
+            xr.DataArray(
+                shift_cftime_index(forecast, 'time', n, freq),
+                dims=['time'],
+                coords=[forecast['time']],
+            )
+            for n in n
+        ],
+        'lead',
+    )
+    # Checks at each `init` if all leads can verify.
+    verifies_at_all_leads = init_lead_matrix.isin(verif['time']).all('lead')
+    forecast = forecast.where(verifies_at_all_leads, drop=True)
+    return forecast, verif
+
+
+def set_cftime_to_int_dim(ds, dim, calendar=PM_CALENDAR_STR):
```

Comment:
Does this need to be in `utils` if it's only being used in tests and is replicating `convert_time_index`? Can it just stay in the test section? I think the name should actually be `set_int_time_to_cftime`. Also, this currently overwrites the file. 

If you do:

```python
from climpred.tutorial import load_dataset
from climpred.utils import set_cftime_to_int_dim
control = load_dataset('MPI-control-1D')
# Works and converts to cftime
set_cftime_to_int_dim(control, 'time')
# Breaks, since it was overwritten.
set_cftime_to_int_dim(control, 'time')
```

So you should probably have a `.copy()` at the top of this somewhere.

Reply:
I guess the only way to actually do this is with `ds = ds.copy()` at the top. `convert_time_index` doesn't have this overwriting problem by doing so.

Reply:
Or we could just have a `utils.py` within the tests folder..

Reply:
Can stay in conftest. Probably I can use converttimeindex

Reply:
can go away. not needed. `convert_time_index` does the same.